### PR TITLE
Fix bug that allowed duplicate sources to be created for a source suggestion

### PR DESCRIPTION
**Pull request by susannasiebert**
The query as it was previously written would always return `nil` because it would search with the source_type string instead of the underlying integer. This would result in a new source to always get created when making a source suggestion, even if that source already existed in our database.

### CODE REVIEW

**Comment by susannasiebert** [comment]
@tmooney Good catch. I found a couple of more places where this occurs and fixed them.

File: app/admin/utilities.rb [hunk 1]
```diff
@@ -143,19 +143,27 @@
...
+    proposed_source_type_int = Source.source_types[proposed_source_type]
     notice = if proposed_citation_id.blank?
                "No Source Id provided."
              elsif proposed_source_type.blank?
                "No Source Type provided."
-             elsif (source = Source.find_by(citation_id: proposed_citation_id, source_type: proposed_source_type))
+             elsif (source = Source.find_by(citation_id: proposed_citation_id, source_type: proposed_source_type_int))
                "Source '#{source.description}' already present."
-             elsif proposed_source_type == 'pubmed'
+             elsif proposed_source_type == 'PubMed'
```
Comment (tmooney) [inline app/admin/utilities.rb]: Bonus fix!

**Comment by susannasiebert** [comment]
@acoffman almost certainly :/

**Comment by acoffman** [comment]
@susannasiebert cool, i'll make an issue to clean them up when we deploy.

**Pull request merged**